.github/workflows/test.yml

```diff
@@ -32,201 +30,23 @@ jobs:
-
-      - name: Checkout kurtosis-cdk
-        uses: actions/checkout@v4
-        with:
-          repository: 0xpolygon/kurtosis-cdk
-          ref: ${{ env.KURTOSIS_CDK_GIT_REF }}
-          path: kurtosis-cdk
-
-      - name: Replace param with local agglayer-rs docker img
-        run: |
-          cd kurtosis-cdk
-          new_value="${{ needs.pre-e2e.outputs.tags }}"
-
-          # Escape special characters in the new value
-          escaped_new_value=$(sed 's/[\/&]/\\&/g' <<< "$new_value")
-
-          sed -i "s/zkevm_agglayer_image: .*/zkevm_agglayer_image: $escaped_new_value/" "params.yml"
-
-          # Debug
-          cat params.yml
-
-      - name: Install kurtosis
-        run: |
-          echo "deb [trusted=yes] https://apt.fury.io/kurtosis-tech/ /" | sudo tee /etc/apt/sources.list.d/kurtosis.list
-          sudo apt update
-          # sudo apt install kurtosis-cli=0.88.19
-          sudo apt install kurtosis-cli
-          kurtosis analytics disable
-
-      - name: Install Foundry
         uses: foundry-rs/foundry-toolchain@v1
 
-      - name: Run kurtosis
-        run: |
-          cd kurtosis-cdk
-          kurtosis run --enclave cdk-v1 --args-file params.yml --image-download always .
-
-      - name: Expose RPC URL
-        run: echo "ETH_RPC_URL=$(kurtosis port print cdk-v1 zkevm-node-rpc-001 http-rpc)" >> $GITHUB_ENV
-
-      - name: Check that batches are progressing
-        continue-on-error: true
-        env:
-          COMMAND: "cast rpc zkevm_batchNumber && cast rpc zkevm_virtualBatchNumber && cast rpc zkevm_verifiedBatchNumber"
-          ETH_RPC_URL: ${{ env.ETH_RPC_URL }}
-          INTERVAL_SECONDS: 10
-          TIMEOUT_SECONDS: 900
-        run: >
-          ./.github/workflows/scripts/are_counters_progressing.sh
-          "$COMMAND"
-          "$TIMEOUT_SECONDS"
-          "$INTERVAL_SECONDS"
-      - name: debug
-        run: kurtosis service logs cdk-v1 zkevm-agglayer-001
-
-  pushing-docker-image:
-    name: Docker | Publish image to registry
-    runs-on: ubuntu-latest
-    permissions:
-      packages: write
-    needs: 
-      - pre-e2e
-      - e2e-kurtosis-cdk
-    steps:
-      - name: Download artifact
-        uses: actions/download-artifact@v4
-        with:
-          name: "docker-image"
-          path: "/tmp"
-
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@v3
-        with:
-          registry: ${{ env.REGISTRY }}
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Load image
-        run: |
-          echo ${{ needs.pre-e2e.outputs.tags }}
-          docker load --input /tmp/docker-image.tar
-          docker image ls -a
-
-      - name: Publishing image
-        env:
-          DOCKER_IMAGE: ${{ needs.pre-e2e.outputs.tags }}
-        run: docker push ${{ env.DOCKER_IMAGE }}
-
-
+      - name: Test
+        run: cargo nextest run --workspace --nocapture && cargo test --doc --workspace
```

Comment:
The `--nocapture` (which is an alias to `--no-capture`) forces the tests to run serially. Is that the intention?

I'd say using Nextest's `--success-output=immediate` and `--failure-output=immediate-final` is preferable here.

[Relevant docs here](https://nexte.st/docs/reporting/#standard-output-and-standard-error)